lib/action_view/component/content_areas.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
+    module ContentAreas
+      extend ActiveSupport::Concern
+
+      included do
+        class_attribute :content_areas, default: []
+        self.content_areas = [] # default doesn't work until Rails 5.2
+      end
+
+      class_methods do
+        def set_content_areas(*areas)
```

Comment:
This is looking like a really powerful enhancement @bostanio. I wonder whether `define_content_areas` or even `has_content_areas` might be slightly more Rails-y than `set_content_areas`; what do you think?

Reply:
@spdawson Yup this always bugged me as not being Rails-y. Just renamed to `with_content_areas` which mirrors nicely the call to `with` that it enables. That's for the inspiration on a better set of names.